### PR TITLE
Pin python package versions and update pips

### PR DESCRIPTION
**Pull request by sd109**
Pin pip-installed versions of pymysql, websockify and pip itself to avoid breakage from new upstream package versions.

### CODE REVIEW

**Comment by sd109** [comment]
Manually tested (via CaaS Slurm appliance branch [here](https://github.com/stackhpc/caas-slurm-appliance/compare/main...fix/ood-pip-update)) but happy to wait for SMS to be fixed so that proper CI can run before merging if preferred.

File: ansible/roles/openondemand/tasks/vnc_compute.yml
Comment (JohnGarbutt) [inline ansible/roles/openondemand/tasks/vnc_compute.yml:30]: shouldn't we pin this to a specific version as well? pip can break things.

File: ansible/roles/openondemand/tasks/vnc_compute.yml
Comment (sjpb) [inline ansible/roles/openondemand/tasks/vnc_compute.yml:30]: Passing a version to `state` is [not documented](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html#parameter-state) as working

**Review by sjpb — CHANGES_REQUESTED** [review]

**Comment by sjpb** [comment]
@sd109 which makes me wonder if there should be a `pip_version` default in environments/common/inventory/group_vars/defaults which everything could use?

**Comment by sd109** [comment]
Yeah probably, I'll make sure this CI actually passes then make that change.

**Comment by sjpb** [comment]
Turns out this was fixed by https://github.com/stackhpc/ansible-slurm-appliance/pull/294 but CaaS hadn't been updated to include that.

**Pull request closed without merging**
